### PR TITLE
Making WhitespacyLinesFixer not change the contents of strings

### PR DESCRIPTION
**Pull request by Sam-Burns**
Fixes problem where regex changes the insides of string variables, resulting in accidental functional change.

### CODE REVIEW

**Comment by stof** [comment]
@keradus removing trailing spaces at the end of string is changing the behavior of the code (dependin gof what the string is). IMO, saying that PSR-2 forces us to remove spaces in this case is wrong. Working code is more important than formatted code

File: Symfony/CS/Fixer/PSR2/TrailingSpacesFixer.php [hunk 1]
```diff
@@ -26,7 +27,19 @@ class TrailingSpacesFixer extends AbstractFixer
...
         // [Structure] Don't add trailing spaces at the end of non-blank lines
-        return preg_replace('/(?<=\S)[ \t]+$/m', '', $content);
+        $contentAfterRegex = preg_replace('/(?<=\S)[ \t]+$/m', '', $content);
+
+        $originalTokens = Tokens::fromCode($content);
+        $newTokens = Tokens::fromCode($contentAfterRegex);
+
+        foreach ($newTokens as $tokenIndex => $newToken) {
+            if ($newToken->isGivenKind(array(T_ENCAPSED_AND_WHITESPACE, T_CONSTANT_ENCAPSED_STRING))) {
+                $tokenBeforeRegex = $originalTokens[$tokenIndex];
```
Comment (stof) [inline Symfony/CS/Fixer/PSR2/TrailingSpacesFixer.php]: this logic looks wrong to me. Can you be sure that the removal of trailing whitespaces does not change anything in the collection indexes ?

**Comment by keradus** [comment]
So @stof, I will say it third time ;)
Yeah, go ahead!
Then we will just need to add a fixer which will remove all trailing whitespaces, because some of user rely on exactly that behavior (want to trim it even if it's a content of string)

It may be tricky to set a level of it. It should be PSR2 but it may change to code, so maybe better CONTRIB.
Or maybe put at PSR2 and create another flag to allow for code behavior change from config.

**Comment by GrahamCampbell** [comment]
Surely both fixers should be PSR-2 level?

**Comment by keradus** [comment]
argh... no, whitespacy lines may contain spaces.

**Comment by GrahamCampbell** [comment]
oh yeh, sorry. derp derp derp :P

**Comment by keradus** [comment]
please calm down

**Comment by GrahamCampbell** [comment]
> please calm down

i was meaning i was derping, lol...

**Comment by keradus** [comment]
It would be really great to have that finished and merged @Sam-Burns !
Any news on that ?

**Comment by keradus** [comment]
It would be really great to have that finished and merged @Sam-Burns !
Any news on that ?

**Comment by keradus** [comment]
Any chances to finish that @Sam-Burns ?

**Comment by keradus** [comment]
Closing, replaced by #1277 and #1224
Thanks you @Sam-Burns for great catch of bugs and test cases.

**Pull request closed without merging**